### DISCUSSION TITLE
Products Endpoint Not Returning Expected Results

### DISCUSSION THREAD

**Discussion by brickbuilderbob** [saleweaver/python-amazon-sp-api#61] [Q&A] [answered]
Hi, I've been successful so far in connecting to the Catalog endpoint, and have been able to parse results. However, when I try to use the Products endpoints, I don't get what look like normal results returned. Below is my code. The ASIN I'm checking is a known ASIN that I can find on Amazon.

from sp_api.api import Products
from sp_api.base import SellingApiException, Marketplaces
import json
import ast

print('Start API')

ASINString = 'B085B26VXX'
Response = Products().get_product_pricing_for_asins(str(ASINString),item_condition = str('New'))
print(Response.payload)

The results I'm getting look like they are parsing each of the individual characters of the "ASINString" I've created and checking each of them. Below is the printed response I'm getting:

[
{'status': 'Success', 'ASIN': 'B', 'Product': {'Identifiers': {'MarketplaceASIN': {'MarketplaceId': 'ATVPDKIKX0DER', 'ASIN': 'B'}}}}
, {'status': 'Success', 'ASIN': '0', 'Product': {'Identifiers': {'MarketplaceASIN': {'MarketplaceId': 'ATVPDKIKX0DER', 'ASIN': '0'}}}}
, {'status': 'Success', 'ASIN': '8', 'Product': {'Identifiers': {'MarketplaceASIN': {'MarketplaceId': 'ATVPDKIKX0DER', 'ASIN': '8'}}}}
, {'status': 'Success', 'ASIN': '5', 'Product': {'Identifiers': {'MarketplaceASIN': {'MarketplaceId': 'ATVPDKIKX0DER', 'ASIN': '5'}}}}
, {'status': 'Success', 'ASIN': '2', 'Product': {'Identifiers': {'MarketplaceASIN': {'MarketplaceId': 'ATVPDKIKX0DER', 'ASIN': '2'}}}}
, {'status': 'Success', 'ASIN': '6', 'Product': {'Identifiers': {'MarketplaceASIN': {'MarketplaceId': 'ATVPDKIKX0DER', 'ASIN': '6'}}}}
, {'status': 'Success', 'ASIN': 'V', 'Product': {'Identifiers': {'MarketplaceASIN': {'MarketplaceId': 'ATVPDKIKX0DER', 'ASIN': 'V'}}}}
, {'status': 'Success', 'ASIN': 'X', 'Product': {'Identifiers': {'MarketplaceASIN': {'MarketplaceId': 'ATVPDKIKX0DER', 'ASIN': 'X'}}}}
]

I'm new to using Python, so I could be making a simple mistake in providing a string as a list, but from what I understand, this is the appropriate way to provide the asin_list argument in this instance. Thanks for your help!

**Comment by saleweaver** [comment, marked as answer]
asins needs to be a list, not a string. The rest of the code looks fine and will work.

**Reply by brickbuilderbob** [reply]
Thanks so much! That worked. I need to read more carefully; it was defined in the documentation as a LIST, not a string.
